toolchain/check/function.cpp

```diff
@@ -191,60 +191,88 @@ auto CheckFunctionTypeMatches(Context& context,
                      context.functions().Get(prev_function_id), substitutions);
 }
 
+// Checks to see if a structurally valid redeclaration is allowed in context.
+// These all still merge.
+static auto CheckIsAllowedRedecl(Context& context, Parse::NodeId node_id,
+                                 SemIR::Function& new_function,
+                                 bool new_is_definition,
+                                 SemIR::Function& prev_function,
+                                 bool prev_is_import) -> void {
+  CARBON_DIAGNOSTIC(FunctionPreviousDecl, Note, "Previously declared here.");
+  if (prev_is_import) {
+    // TODO: Allow non-extern declarations in the same library.
+    if (!new_function.is_extern && !prev_function.is_extern) {
+      CARBON_DIAGNOSTIC(
+          FunctionNonExternRedecl, Error,
+          "Only one library can declare function {0} without `extern`.",
+          SemIR::NameId);
+      context.emitter()
+          .Build(node_id, FunctionNonExternRedecl, prev_function.name_id)
+          .Note(prev_function.decl_id, FunctionPreviousDecl)
+          .Emit();
+      return;
+    }
+  } else {
```

Comment:
(Optional, no action needed.)

```suggestion
    }
    return;
  }
```

Could maybe reduce indentation here.

Reply:
I feel kind of odd about reducing the indent this way. It has an odd implication to me that imported versus non-imported is more significant; the `else` was intended to treat the cases as equal.